packages/block-library/src/gallery/index.php

```diff
@@ -32,6 +32,43 @@ function block_core_gallery_data_id_backcompatibility( $parsed_block ) {
 
 add_filter( 'render_block_data', 'block_core_gallery_data_id_backcompatibility' );
 
+/**
+ * Adds a style tag for the --wp--style--unstable-gallery-gap var.
+ *
+ * The Gallery block needs to recalculate Image block width based on
+ * the current gap setting in order to maintain the number of flex columns
+ * so a css var is added to allow this.
+ *
+ * @param array  $attributes Attributes of the block being rendered.
+ * @param string $content Content of the block being rendered.
+ * @return string The content of the block being rendered.
+ */
+function block_core_gallery_render( $attributes, $content ) {
+	$gap_value = _wp_array_get( $attributes, array( 'style', 'spacing', 'blockGap' ) );
+	// Skip if gap value contains unsupported characters.
+	// Regex for CSS value borrowed from `safecss_filter_attr`, and used here
+	// because we only want to match against the value, not the CSS attribute.
+	$gap_value = preg_match( '%[\\\(&=}]|/\*%', $gap_value ) ? null : $gap_value;
+	$id        = uniqid();
+	$class     = 'wp-block-gallery-' . $id;
+	$content   = preg_replace(
+		'/' . preg_quote( 'class="', '/' ) . '/',
+		'class="' . $class . ' ',
+		$content,
+		1
+	);
+	$style     = '.' . $class . '{ --wp--style--unstable-gallery-gap: ' . $gap_value . '}';
```

Comment:
Should we bother rendering the CSS var if `$gap_value` is `NULL`?

Reply:
Or maybe set a default here or in the style sheets?

Reply:
I don't know why, but an empty value for `--wp--style--unstable-gallery-gap` causes the frontend to bork.

https://user-images.githubusercontent.com/6458278/150901993-c402975d-930e-4e55-be41-fe92f9a8e61a.mp4

Reply:
>Or maybe set a default here or in the style sheets

Good catch - we should be setting some defaults here.

Reply:
@ramonjd the same default value should now be set in the editor and front end based on `--wp--style--block-gap` or a fall back to `0.5em` the same as the layout code. The `--wp--style--block-gap` is getting picked up from theme.json as expected, but just from the global gap value - but as you note it probably makes sense to check for one at the block level also, will have a look and see what it possible there.

Reply:
Nice!

> but as you note it probably makes sense to check for one at the block level also, will have a look and see what it possible there

Yeah I just noted it so I wouldn't forget. It might have fallen down the crack during the removal of var globally. Not a blocker for this PR I think.